sklearn/tree/_export.py

```diff
@@ -739,6 +725,26 @@ def recurse(self, node, tree, ax, max_x, max_y, depth=0):
             ax.annotate("\n  (...)  \n", xy_parent, xy, **kwargs)
 
 
+@validate_params(
+    {
+        "decision_tree": [DecisionTreeClassifier, DecisionTreeRegressor],
```

Comment:
I don't think it's only these two, and I think third part libraries also use this method to visualize trees which are not ours.

```py
        # Now recurse the tree and add node & edge attributes
        if isinstance(decision_tree, _tree.Tree):
            self.recurse(decision_tree, 0, criterion="impurity")
        else:
            self.recurse(decision_tree.tree_, 0, criterion=decision_tree.criterion)
```

And `recurse` doesn't check the type, so anything compatible with our types can actually work here.

Reply:
@adrinjalali Thanks for your review. So do you recommend just leave it as "no_validation"?

Reply:
It's not strictly required to be one of (or a subclass of one of) these 2 classes so yes let's put "no_validation"

Reply:
yeah I think no_validation is fine, since actual validation would be quite hard, and `no_validation` is in effect what we have currently anyway.